extensions/gitpod-web/src/extension.ts

```diff
@@ -397,27 +397,29 @@ export class GitpodWorkspaceTreeDataProvider implements vscode.TreeDataProvider<
 				return;
 			}
 			const toClean = new Set<number>(this.ports.ports.keys());
-			for (const portStatus of this.portStatus.getPortsList()) {
-				const portNumber = portStatus.getLocalPort();
-				toClean?.delete(portNumber);
-				const port = this.ports.ports.get(portNumber) || new GitpodWorkspacePort(portNumber, this.context);
-				this.ports.ports.set(portNumber, port);
-				const currentStatus = port.status;
-				port.status = portStatus.toObject();
-				port.tunnel = this.tunnels.get(portNumber);
-				const labelPrefix = currentStatus?.name ? `${currentStatus.name}: ` : '';
-				port.label = labelPrefix + portNumber;
-				if (currentStatus?.description) {
-					port.tooltip = `${port.label} - ${currentStatus.description}`;
+			const portsList = this.portStatus.getPortsList();
+			for (const portStatus of portsList) {
+				const currentStatus = portStatus.toObject();
+				const { name, localPort, description, exposed, served } = currentStatus;
+				toClean?.delete(localPort);
+
+				const port = this.ports.ports.get(localPort) || new GitpodWorkspacePort(localPort, this.context);
+				const prevStatus = port.status;
+				this.ports.ports.set(localPort, port);
+
+				port.status = currentStatus;
+				port.tunnel = this.tunnels.get(localPort);
+				port.label = name ? `${name}: ${localPort}` : `${localPort}`;
+				if (description) {
+					port.tooltip = name ? `${name} - ${description}` : description;
 				}
-				const remotePort = port.remotePort;
-				if (remotePort && remotePort !== portNumber) {
-					port.label += ':' + remotePort;
+				if (port.remotePort && port.remotePort !== localPort) {
+					port.label += ':' + port.remotePort;
 				}
 
-				const exposed = portStatus.getExposed();
 				const accessible = exposed || port.tunnel;
-				if (!portStatus.getServed()) {
+				const isPortTunnelPublic = port.tunnel?.privacy === 'public'; // TunnelPrivacyId
```

Comment:
I don't know how to import `TunnelPrivacyId` here, code below not works
```
import { TunnelPrivacyId } from 'vs/platform/tunnel/common/tunnel';
```

Reply:
The values come from [here](https://github.com/gitpod-io/openvscode-server/blob/3dd5bb05411a5eb2a9ded6aca5d38d938fefaf7f/src/vs/gitpod/browser/workbench/workbench.ts#L801-L812) but seems we don't use it, only `public` [here](https://github.com/gitpod-io/openvscode-server/blob/3dd5bb05411a5eb2a9ded6aca5d38d938fefaf7f/src/vs/gitpod/browser/workbench/workbench.ts#L790-L794) so just continue using `public` :smile: